Plugins/Public/alley/PlayerRestrictions.cpp

```diff
@@ -718,8 +719,23 @@ bool UserCmd_JettisonAll(uint iClientID, const wstring &wscCmd, const wstring &w
 				if (skipItem) {
 					continue;
 				}
-				HkRemoveCargo(wscCharname, item->iID, item->iCount);
-				Server.MineAsteroid(iSystem, vLoc, CreateID("lootcrate_ast_loot_metal"), item->iArchID, item->iCount, iClientID);
+
+				if (isFirstJettisonItem)
+				{
+					isFirstJettisonItem = false;
+					XJettisonCargo jettisonCargo;
+					jettisonCargo.iShip = iShip;
+					jettisonCargo.iCount = item->iCount;
+					jettisonCargo.iSlot = item->iID;
+					pub::Player::SendNNMessage(iClientID, pub::GetNicknameId("cargo_jettisoned"));
+					Server.JettisonCargo(iClientID, jettisonCargo);
```

Comment:
is it implied here that this is performing HkRemoveCargo (or rather the equivalent thereof) as well behind the scenes?

Reply:
Yes, it performs the full jettison sequence, removal of cargo, jettisoning it into space, playing cargo hold animation and disabling shields temporarily. Only the 'cargo jettison' needs to be played manually